common/warp_utils/src/cors.rs

```diff
@@ -0,0 +1,21 @@
+use std::net::Ipv4Addr;
+use warp::filters::cors::Builder;
+
+/// Configure a `cors::Builder`.
+///
+/// If `allow_origin.is_none()` the `default_origin` is used.
+pub fn set_builder_origins(
+    builder: Builder,
+    allow_origin: Option<&str>,
+    default_origin: (Ipv4Addr, u16),
+) -> Builder {
+    if let Some(allow_origin) = allow_origin {
+        if allow_origin == "*" {
+            builder.allow_any_origin()
+        } else {
+            builder.allow_origins(allow_origin.split(','))
```

Comment:
I tried inputting an invalid origin, and `allow_origins` panics and shuts down the http server.  So we should probably add our own validation.

Reply:
You're right, that will make for nicer UX. I've added the verification here: https://github.com/sigp/lighthouse/pull/1772/commits/d070a7c13c8619c37ce626536101595eed1a8c64.